jobs/bbr-atcdb/templates/connection_data.erb

```diff
@@ -0,0 +1,188 @@
+    postgres_tls_private_key = ""
+
+    if postgres_tls_enabled
+      postgres_tls_skip_host_verify = p("postgresql.tls.skip_host_verify", false)
+      postgres_tls_ca = p("postgresql.tls.cert.ca")
+      postgres_tls_public_cert = p("postgresql.tls.cert.certificate")
+      postgres_tls_private_key = p("postgresql.tls.cert.private_key")
+    end
+
+    if postgres_host.empty?
+      if_link("concourse_db") do |cdb|
+        postgres_host = cdb.p("postgresql.host", "")
+      end
+
+      if postgres_host.empty?
+        if_link("postgres") do |postgres|
+          postgres_host = postgres.instances.first.address
+        end
+      end
+
+      if postgres_host.empty?
+        raise "postgres.host not found through either properties or links"
+      end
+    end
+
+    if postgres_database.empty?
+      if_link("concourse_db") do |cdb|
+        postgres_database = cdb.p("postgresql.database", "")
+      end
+
+      if postgres_database.empty?
+        postgres_database = "atc"
+      end
+    end
+
+    if postgres_port.zero?
+      if_link("concourse_db") do |cdb|
+        postgres_port = cdb.p("postgresql.port", 0)
+      end
+
+      if postgres_port.zero?
+        if_link("postgres") do |postgres|
+          postgres_port = postgres.p("databases.port", 0)
+        end
+      end
+
+      if postgres_port.zero?
+        postgres_port = 5432
+      end
+    end
+
+    if postgres_role_name.empty?
+      if_link("concourse_db") do |cdb|
+        postgres_role_name = cdb.p("postgresql.role.name", "")
+      end
+
+      if postgres_role_name.empty?
+        raise "postgres.role.name not found through either properties or links"
+      end
+    end
+
+    if postgres_role_password.empty?
+      if_link("concourse_db") do |cdb|
+        postgres_role_password = cdb.p("postgresql.role.password", "")
+      end
+
+      if postgres_role_password.empty?
+        raise "postgres.role.password not found through either properties or links"
+      end
+    end
+
+    if_link("concourse_db") do |cdb|
+      postgres_tls_enabled = cdb.p("postgresql.sslmode", postgres_tls_enabled)
+      if postgres_tls_enabled
+        if postgres_tls_ca.empty?
+          postgres_tls_ca = cdb.p("postgresql.ca_cert.ca")
+        end
+
+        if postgres_tls_public_cert.empty?
+          postgres_tls_public_cert = cdb.p("postgresql.client_cert.certificate")
+        end
+
+        if postgres_tls_private_key.empty?
+          postgres_tls_private_key = cdb.p("postgresql.client_cert.private_key")
+        end
+      end
+    end
+-%>
+
+
+<%
+  connection_string = "host="+postgres_host \
+    +" user="+postgres_role_name \
+    +" password="+postgres_role_password \
+    +" port="+postgres_port.to_s() \
+    +" dbname="+postgres_database
+
+    sslmode = "disable"
+
+    unless postgres_tls_skip_host_verify
+      sslmode = "require"
+
+      unless postgres_tls_ca.empty?
+-%>
+<%=        env_file_writer(postgres_tls_ca, "POSTGRES_TLS_CA") %>
```

Comment:
hm not sure why this line and others like it got generated in it's own block. Cause it's just a function call?

Reply:
the `<%= %>` block expects only one value to be output. So if we put multiple env_file_writer in one block it would be just one line be generated in the end for creating a key file. See details [here](https://puppet.com/docs/puppet/7.5/lang_template_erb.html#expression-printing-tags)

we have similar code 
https://github.com/concourse/concourse-bosh-release/blob/master/jobs/web/templates/pre_start.erb#L48-L50